packages/medusa/src/api/admin/promotions/validators.ts

```diff
@@ -125,15 +125,14 @@ export type AdminUpdateApplicationMethodType = z.infer<
 >
 export const AdminUpdateApplicationMethod = z
   .object({
+    id: z.string(),
```

Comment:
todo: this will break the admin promotion update in dashboard as we don't pass the id in there and this change will require it now

Reply:
I think so, because it will break promotion update functionalities in admin.

An easier way would be to remove this and [update this instead](https://github.com/medusajs/medusa/blob/develop/packages/core/types/src/promotion/common/promotion.ts#L141) to:

```
application_method?: Omit<UpdateApplicationMethodDTO, "id">
```

Reply:
But would this update the application method without its ID?

Reply:
yes, its a 1:1 relationship. The module picks it up from the promotion id that's already passed to its parent. This is only used as a subresource an never referenced directly

Reply:
cool, updated it